sage-ios/SAGE/SAGE/CheckinViewController.swift

```diff
@@ -133,33 +141,20 @@ class CheckinViewController: SGViewController {
                 preferredStyle: .Alert)
             alertController.addAction(UIAlertAction(title: "OK", style: .Default, handler: nil))
             self.presentViewController(alertController, animated: true, completion: nil)
-
+            
         } else if CLLocationManager.locationServicesEnabled() {
             switch CLLocationManager.authorizationStatus() {
             case .AuthorizedWhenInUse, .AuthorizedAlways:
                 // Verify location
-                if self.currentLocationIsBySchool() {
-                    let alertController = UIAlertController(
-                        title: "Start mentoring session?",
-                        message: nil,
-                        preferredStyle: .ActionSheet)
-                    alertController.addAction(UIAlertAction(title: "Start Session", style: .Default, handler: { (action: UIAlertAction) -> Void in
-                        //save start time locally and start timer
-                        self.presentSessionMode(UIConstants.normalAnimationTime)
-                        self.inSession = true
-                        self.startTime = NSDate.timeIntervalSinceReferenceDate()
-                        self.updateSessionTime()
-                        KeychainWrapper.setString(String(format: "%f", self.startTime), forKey: KeychainConstants.kSessionStartTime)
-                    }))
-                    alertController.addAction(UIAlertAction(title: "Cancel", style: .Cancel, handler: nil))
-                    self.presentViewController(alertController, animated: true, completion: nil)
+                if !self.currentLocation.recent() {
+                    self.checkinView.presentActivityIndicator()
+                    self.startGettingCurrentLocation()
+                    semaphore = dispatch_semaphore_create(0)
+                    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)
```

Comment:
do we want to do this? it might be a bit jarring to force the user to wait for the location to update and prevent them from doing anything else. would the user be able to switch between tabs?